metaflow_extensions/plugins/install_step_decorator.py

```diff
@@ -49,11 +77,8 @@ def wrapped_prepare_step_environment(self, *args, **kwargs):
         for pkg_spec in filter(None, map(get_pkg_spec, PREINSTALL_PKGS)):
             pip_install(python_exec, pkg_spec)
 
-        # Install the local project if it conforms to the expected structure
-        flow_path = Path(sys.argv[0])
-        project_root = up_to_project_root(flow_path)
-        if project_root:
-            pip_install(python_exec, project_root)
+        # Install the local project into the current conda env
+        install_flow_project(python_exec)
         return env_id
 
     return wrapped_prepare_step_environment
```

Comment:
Lines 87-89 (can't select those)...

Should the patching be protected by some kind of check?

We probably don't want to patch if `METAFLOW_DEFAULT_ENVIRONMENT` isn't `project`?

The behaviour for installing `PREINSTALL_PKGS` is coupled to the "project" environment, `@install_project` can actually act independently of the "project" environment.

This also suggests that this funcitonality should be moved to `project_environment.py`?

Reply:
yep, fair, there was little reason in the end to house this in `install_step_decorator` at all, just an early design decision that stuck